bc_obps/registration/models.py

```diff
@@ -142,6 +143,13 @@ class Meta:
         models.Index(fields=["type"], name="document_type_idx"),
     ]
 
+    def delete(self, *args, **kwargs):
+        # Delete the file from Google Cloud Storage before deleting the model instance
+        if self.file:
+            default_storage.delete(self.file.name)
+
+        super().delete(*args, **kwargs)
+
```

Comment:
Let's confirm this with @dleard . I feel that hard deleting is not what we should do here since the `Document` data model is a `TimeStampedModel` and I assume we should track its trail. Sorry for the confusion earlier.